### PR TITLE
[android] Kotlinize kernel services

### PR DESCRIPTION
**Pull request by wschurman**
# Why

Part 2 of https://github.com/expo/expo/pull/13685.

# How

- For each file in kernel/services, convert to kotlin, manually inspect conversion (compare against java).
- `gradlew spotlessApply`
- Remove `ClearExperienceData` since it's unused (grep)

# Test Plan

Run native component list https://expo.io/@community/native-component-list in simulator, ensure sensors work.

### CODE REVIEW

File: android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SensorKernelServiceSubscription.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+package host.exp.exponent.kernel.services.sensors
+
+import host.exp.exponent.kernel.ExperienceKey
+
+class SensorKernelServiceSubscription internal constructor(
+  val experienceKey: ExperienceKey,
```
Comment (ide) [inline android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SensorKernelServiceSubscription.kt:7]: Is the setter for this private?

File: android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SensorKernelServiceSubscription.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    assertSubscriptionIsAlive()
+    if (!_isEnabled) {
+      _isEnabled = true
+      subscribableSensorKernelService.onSubscriptionEnabledChanged(this)
+    }
+  }
+
+  private var _isEnabled = false
+  val isEnabled: Boolean
+    get() = !hasBeenReleased && _isEnabled
```
Comment (ide) [inline android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SensorKernelServiceSubscription.kt:25]:
```suggestion
  val isEnabled: Boolean
    get() = !hasBeenReleased && field
```
Or is it too confusing that

this.isEnabled = true
this.isEnabled == false

is possible? (What's idiomatic in Kotlin?)

File: android/expoview/src/main/java/host/exp/exponent/modules/ClearExperienceData.java [hunk 3]
```diff
@@ -1,39 +0,0 @@
-// Copyright 2015-present 650 Industries. All rights reserved.
```
Comment (ide) [inline android/expoview/src/main/java/host/exp/exponent/modules/ClearExperienceData.java:1]: Should there be a .kt complement of this file?

File: android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SubscribableSensorKernelService.kt [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+import android.content.Context
+import android.hardware.SensorEvent
+import host.exp.exponent.kernel.ExperienceKey
+import java.lang.ref.WeakReference
+import java.util.*
+
+private var DEFAULT_UPDATE_INTERVAL = 100
+
+abstract class SubscribableSensorKernelService internal constructor(reactContext: Context) : BaseSensorKernelService(reactContext) {
+  private val experienceScopeKeyListenersCountMap: MutableMap<String?, Int> = mutableMapOf()
```
Comment (ide) [inline android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SubscribableSensorKernelService.kt:13]: The null keys look sketchy but I see the comments below. At least Kotlin makes this code explicitly sketchy

**Review by ide — CHANGES_REQUESTED** [review]

File: android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SensorKernelServiceSubscription.kt (hunk 1 above)
Comment (wschurman) [reply android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SensorKernelServiceSubscription.kt:7]: it's a `val` so no setter

File: android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SensorKernelServiceSubscription.kt (hunk 2 above)
Comment (wschurman) [reply android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SensorKernelServiceSubscription.kt:25]: yeah, for now I think a backing field is best to limit scope of this PR, but a refactor is definitely possible to make this a bit cleaner. I do think it's a bit odd to use `field` and something else in the getter.

File: android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SubscribableSensorKernelService.kt (hunk 4 above)
Comment (wschurman) [reply android/expoview/src/main/java/host/exp/exponent/kernel/services/sensors/SubscribableSensorKernelService.kt:13]: Yeah, I tried to remove the nullable key, but it'll take a follow-up PR adding a null check somewhere at the callsites to fix.

File: android/expoview/src/main/java/host/exp/exponent/modules/ClearExperienceData.java (hunk 3 above)
Comment (wschurman) [reply android/expoview/src/main/java/host/exp/exponent/modules/ClearExperienceData.java:1]: Nope, this file was unused so I removed it. Checked for usage via grep.

**Review by ide — APPROVED** [review]

**Pull request merged**